x-pack/plugins/ml/server/models/json_schema_service/schema_overrides.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SupportedPath } from '../../routes/schemas/json_schema_schema';
+import type { PropertyDefinition } from './types';
+
+/**
+ * Extension of the schema definition extracted from the Elasticsearch specification.
+ */
+export const jsonSchemaOverrides: Record<SupportedPath, Partial<PropertyDefinition>> = {
+  '/_ml/anomaly_detectors/{job_id}': {
+    // background_persist_interval is required according to the ES spec
+    required: ['analysis_config', 'data_description'],
+    properties: {
+      job_id: {
```

Comment:
it's interesting that you need to add this when then type `MlJob` from `estypes` contains `job_id`, and I assumed that comes from the same core specification.

Reply:
from the API endpoint perspective, `job_id` is a path component and not one of the request body properties

Reply:
Is the spec that you're using different to the one which is used to create this `MlJob` interface from the elasticsearch js library?

```ts
export interface MlJob {
    allow_lazy_open: boolean;
    analysis_config: MlAnalysisConfig;
    analysis_limits?: MlAnalysisLimits;
    background_persist_interval?: Duration;
    blocked?: MlJobBlocked;
    create_time?: DateTime;
    custom_settings?: MlCustomSettings;
    daily_model_snapshot_retention_after_days?: long;
    data_description: MlDataDescription;
    datafeed_config?: MlDatafeed;
    deleting?: boolean;
    description?: string;
    finished_time?: DateTime;
    groups?: string[];
    job_id: Id;
    job_type?: string;
    job_version?: VersionString;
    model_plot_config?: MlModelPlotConfig;
    model_snapshot_id?: Id;
    model_snapshot_retention_days: long;
    renormalization_window_days?: long;
    results_index_name: IndexName;
    results_retention_days?: long;
}
```

Reply:
Perhaps it's the same one that is used to create `MlPutJobRequest`?
But in `MlPutJobRequest`, `analysis_config` and `data_description` are non-optional.

Reply:
Just checked, and it's actually part of the request, but the `description` property is missing. So I decided to add it to the overrides.